doc/command_ref.rst

```diff
@@ -411,10 +424,49 @@ Install Command
 ---------------
 
 ``dnf [options] install <spec>...``
-    DNF makes sure that the given packages and their dependencies are installed on the system. Each ``<spec>`` can be
-    either a ``<package-spec>``, which specifies a package directly, or a path to the local rpm package, or a URL to a remote rpm package, or a ``@<group-spec>``, which specifies an (environment) group which contains it. If a given package cannot be (and is not already) installed, the exit code will be non-zero.
+    DNF makes sure that the given packages and their dependencies are installed
+    on the system. Each ``<spec>`` can be either a :ref:`<package-spec>
+    <specifying_packages-label>`, or a \@\ :ref:`\<group-spec>\ <specifying_groups-label>`.
+    If a given package or provide cannot be (and is not already) installed,
+    the exit code will be non-zero.
+
+    When :ref:`<package-spec> <specifying_packages-label>` that specify exact version
+    of the package is given, DNF will install the desired version, no matter which
+    version of the package is already installed. The former version of the package
+    will be removed in the case of non-installonly package.
+
+    Please make sure that you understand which package will be selected in case
+    of multiple matches (see :ref:`\configuration_files_replacement_policy-label`).
+
+Examples
+--------
+
+``dnf install tito``
+    Install package tito (tito is package name).
+
+``dnf install ~/Downloads/tito-0.6.2-1.fc22.noarch.rpm``
+    Install local rpm file tito-0.6.2-1.fc22.noarch.rpm in ~/Downloads/
+    directory.
 
-    Please make sure that you understand which package will be selected in case of multiple matches (see :ref:`\specifying_packages-label`). See also :ref:`\configuration_files_replacement_policy-label`.
+``dnf install tito-0.5.6-1.fc22``
+    Install package with specific version. If the package is already installed it
+    will automatically try to downgrade or upgrade to specific version.
+
+``dnf --best install tito``
+    Install the latest available version of package. If the package is already installed it
+    will automatically try to upgrade to the latest version. If the latest version
+    of package cannot be installed, the installation fail.
+
+``dnf install vim``
+    If you use the command, DNF will automatically recognize that vim is not a
```

Comment:
DNF will automatically recognize that vim is not a package name, but provide, and install a package that provides vim with all required dependencies. Note package name match has precedence over package provides matches.